xds/internal/resolver/xds_resolver_test.go

```diff
@@ -250,6 +281,95 @@ func waitForWatchRouteConfig(ctx context.Context, t *testing.T, xdsC *fakeclient
 	}
 }
 
+// TestXDSResolverResourceNameToWatch tests that the correct resource name is
+// used to watch for the service. This covers cases with different bootstrap
+// config, and different authority.
+func (s) TestXDSResolverResourceNameToWatch(t *testing.T) {
+	tests := []struct {
+		name   string
+		bc     *bootstrap.Config
+		target *resolver.Target
```

Comment:
Same comment about using `Dial()` with a target string, instead of directly calling `Build()`.

Reply:
Replied to the other comment.